src/query/storages/fuse/src/pruning/fuse_pruner.rs

```diff
@@ -243,12 +285,39 @@ impl FusePruner {
                     }
 
                     let mut res = vec![];
+                    let mut deleted_segments = vec![];
                     let pruned_segments = segment_pruner.pruning(batch).await?;
-                    for (location, info) in pruned_segments {
-                        res.extend(block_pruner.pruning(location, &info).await?);
-                    }
 
-                    Result::<_, ErrorCode>::Ok(res)
+                    if delete_purning {
+                        // inverse purn
+                        for (segment_location, compact_segment_info) in &pruned_segments {
+                            // for delete_purn
+                            if !inverse_range_index
+                                .as_ref()
+                                .unwrap()
```

Comment:
shall we just pass in an `Option<RangIndex>`, then match it? so that unwrap can be avoided?